pkg/controller/podautoscaler/controller_metrics/metric_recorder.go

```diff
@@ -0,0 +1,17 @@
+package metrics
```

Comment:
Since the "metrics" term is also part of our domain I would avoid using metrics here. How about `monitoring` or`monitor`? This way we avoid ambiguous `metrics` imported name in `horizontal.go`.

If you agree, then:
* The path could change into `pkg/controller/podautoscaler/monitoring/monitor.go`
* `MetricsRecorder` could change into `ReconciliationMonitor` (alternatively just `Monitor`)
* `ObserveReconciliationResult` could change into `RecordResult` (if `Monitor` above then `RecordReconciliationResult`)